core/src/main/java/com/datastax/oss/driver/internal/core/metadata/LoadBalancingPolicyWrapper.java

```diff
@@ -180,14 +181,17 @@ private void processNodeStateEvent(NodeStateEvent event) {
         return; // ignore
       case RUNNING:
         for (LoadBalancingPolicy policy : policies) {
-          if (event.newState == NodeState.UP) {
-            policy.onUp(event.node);
+          DefaultNode node = event.node.get();
+          if (node == null) {
+            LOG.info("[{}] Node for this event was removed, ignoring: {}", logPrefix, event);
```

Comment:
Info-level log might be a bit high for this notice as there isn't really action the user should take when this happens. Consider dropping to debug/trace